tests/unit/api/test_tasks.py

```diff
@@ -72,3 +74,105 @@ def test_get_result_is_exception(self, test_client, monkeypatch):
         assert mocked_repository_metadata.AsyncResult.calls == [
             pretend.call("test_id")
         ]
+
+    def test_get_result_is_errored(self, test_client, monkeypatch):
+        mocked_task_result = pretend.stub(
+            state="SUCCESS",
+            result={
+                "status": False,
+                "task": "sign_metadata",
+                "last_update": "2023-11-17T09:54:15.762882",
+                "message": "Signature Failed",
+                "error": "No signatures pending for root",
+                "details": {},
+            },
+        )
+        mocked_repository_metadata = pretend.stub(
+            AsyncResult=pretend.call_recorder(lambda t: mocked_task_result)
+        )
+        monkeypatch.setattr(
+            "repository_service_tuf_api.tasks.repository_metadata",
+            mocked_repository_metadata,
+        )
+
+        test_response = test_client.get(f"{TASK_URL}?task_id=test_id")
+        assert test_response.status_code == status.HTTP_200_OK
+
+        assert test_response.json() == {
+            "data": {
+                "task_id": "test_id",
+                "state": "ERRORED",
+                "result": {
+                    "status": False,
+                    "task": "sign_metadata",
+                    "last_update": "2023-11-17T09:54:15.762882",
+                    "message": "Signature Failed",
```

Comment:
Why do we need to have a `message` attribute at all when the state is clearly `ERRORED`?

Reply:
This is related to the changes where we move the `error` and `message` out from `details` to the task result

It is correct message is required.  Error is optional, actually it is required when `status` is `False` (it is also another PR in the worker for that)
Here is an example:
 
 https://github.com/repository-service-tuf/repository-service-tuf-worker/blob/486e98cdbc16dd41c62c0ec8b473e2f94312ee9c/repository_service_tuf_worker/repository.py#L625-L632

Reply:
Okay, let's leave it then.